lib/Sema/CSSimplify.cpp

```diff
@@ -9247,6 +9247,14 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyTransitivelyConformsTo(
   if (resolvedTy->isTypeVariableOrMember())
     return formUnsolved();
 
+  // There is an implicit conversion between `CGFloat` and `Double` types
+  // so if `CGFloat` argument doesn't satisfy the requirement it's possible
+  // that the generic parameter it's passed to be inferred as a `Double`
+  // from context and the requirement to be satisfied through this implicit
+  // conversion.
+  if (resolvedTy->isCGFloat())
+    return SolutionKind::Solved;
```

Comment:
Isn't this also an issue going from Double to CGFloat? e.g:

```swift
import CoreGraphics

protocol P {}
extension CGFloat: P {}

struct S<T> {}

func foo<T: P>(_ x: T) -> S<T> { .init() }

func bar(_ x: Double, _ y: CGFloat) {
  _ = {
    let x: S<CGFloat> = foo(x)
  }
}
```

Reply:
Yeah, I didn't include it because I'm afraid to cause perf regressions because `Double` is more widely used.

Reply:
Let's try that as a separate commit and we can revert if that doesn't pan out.